src/Nethermind/Nethermind.Trie/TrieNode.Visitor.cs

```diff
@@ -97,7 +114,7 @@ internal void Accept(ITreeVisitor visitor, ITrieNodeResolver nodeResolver, TrieV
                 case NodeType.Leaf:
                 {
                     visitor.VisitLeaf(this, trieVisitContext, Value);
-                    if (!trieVisitContext.IsStorage && trieVisitContext.ExpectAccounts) // can combine these conditions
+                    if (!trieVisitContext.IsStorage && visitor.VisitAccounts) // can combine these conditions
```

Comment:
Broken pattern.

Reply:
Pattern is some guidance, its not something that should constrain us if we have similar but more complex use-case.